maven-model-builder/src/main/java/org/apache/maven/model/profile/DefaultProfileActivationContext.java

```diff
@@ -194,15 +162,30 @@ public Map<String, String> getProjectProperties() {
     }
 
     public DefaultProfileActivationContext setProjectProperties(Properties projectProperties) {
-        if (projectProperties != null) {
-            this.projectProperties = projectProperties.entrySet().stream()
-                    .collect(collectingAndThen(
-                            toMap(k -> String.valueOf(k.getKey()), v -> String.valueOf(v)),
-                            Collections::unmodifiableMap));
-        } else {
-            this.projectProperties = Collections.emptyMap();
-        }
+        return setProjectProperties(toMap(projectProperties));
+    }
+
+    public DefaultProfileActivationContext setProjectProperties(Map<String, String> projectProperties) {
+        this.projectProperties = unmodifiable(projectProperties);
```

Comment:
Previously, a `null` for the `projectProperties` argument would result in `this.projectProperties` being an empty `Map`. With this patch, it would result in `this.projectProperties` being `null`. Are you sure that's an intended change?

Reply:
> Previously, a `null` for the `projectProperties` argument would result in `this.projectProperties` being an empty `Map`. With this patch, it would result in `this.projectProperties` being `null`. Are you sure that's an intended change?

I don't think that's true.  `toMap` can return `null`, but `unmodifiable(x)` returns an empty list/map, so `this.projectProperties` can never be `null` afaik.

Reply:
Ah, I see, we now have *two* methods named `setProjectProperties`. The `null` guarding is now split over those two methods, where it was previously in one method.